internal/consumer/partitionConsumer.go

```diff
@@ -72,20 +71,19 @@ func newPartitionConsumer(
 	maxUnAcked := options.Concurrency + options.RcvBufferSize + 1
 	name := fmt.Sprintf("%v-partition-%v", pConsumer.Topic(), pConsumer.Partition())
 	return &partitionConsumer{
-		id:           pConsumer.Partition(),
-		topic:        pConsumer.Topic(),
-		sarama:       sarama,
-		pConsumer:    pConsumer,
-		limit:        limit,
-		reachedLimit: 0,
-		options:      options,
-		msgCh:        msgCh,
-		dlq:          dlq,
-		tally:        scope.Tagged(map[string]string{"partition": strconv.Itoa(int(pConsumer.Partition()))}),
-		logger:       logger.With(zap.String("topic", pConsumer.Topic()), zap.Int32("partition", pConsumer.Partition())),
-		stopC:        make(chan struct{}),
-		ackMgr:       newAckManager(maxUnAcked, scope, logger),
-		lifecycle:    util.NewRunLifecycle(name),
+		id:                pConsumer.Partition(),
+		topic:             pConsumer.Topic(),
+		sarama:            sarama,
+		pConsumer:         pConsumer,
+		decodeDLQMetadata: decodeMetadata,
+		options:           options,
+		msgCh:             msgCh,
+		dlq:               dlq,
```

Comment:
if the partition type is DLQ, make sure dlq is nil i.e no dlq for a dlq consumer

Reply:
This is not necessary. Since we expose Ack/Nack semantics for DLQ as well, it is perfectly valid to nack a message on DLQ merge. This is how a user might achieve more granular DLQ merge because we currently only support range merge.